tests/appsec/test_automated_login_events.py

```diff
@@ -0,0 +1,179 @@
+# Unless explicitly stated otherwise all files in this repository are licensed under the the Apache License Version 2.0.
+# This product includes software developed at Datadog (https://www.datadoghq.com/).
+# Copyright 2022 Datadog, Inc.
+
+from utils import weblog, interfaces, context, missing_feature, released, scenarios
+
+
+@released(cpp="?", golang="?", java="?", nodejs="?", dotnet="?", php="?", ruby="?")
+class Test_Login_Events:
+    "Test login success/failure use cases"
+    # User entries in the internal DB:
+    # users = [
+    #     {
+    #         id: '1',
+    #         username: 'test',
+    #         password: '1234',
+    #         email: 'testuser@ddog.com'
+    #     },
+    #     {
+    #         id: '591dc126-8431-4d0f-9509-b23318d3dce4',
+    #         username: 'testuuid',
+    #         password: '1234',
+    #         email: 'testuseruuid@ddog.com'
+    #     }
+    # ]
+
+    USER = "test"
+    UUID_USER = "testuuid"
+    PASSWORD = "1234"
+    INVALID_USER = "invalidUser"
+
+    BASIC_AUTH_USER_HEADER = "Basic dGVzdDoxMjM0"  # base64(test:1234)
+    BASIC_AUTH_USER_UUID_HEADER = "Basic dGVzdHV1aWQ6MTIzNA=="  # base64(testuuid:1234)
+    BASIC_AUTH_INVALID_USER_HEADER = "Basic aW52YWxpZFVzZXI6MTIzNA=="  # base64(invalidUser:1234)
+    BASIC_AUTH_INVALID_PASSWORD_HEADER = "Basic dGVzdDoxMjM0NQ=="  # base64(test:12345)
+
+    def setup_login_pii_success(self):
+        self.library_name = context.library
+        self.r_pii_success = []
+
+        if self.library_name == "nodejs":
+            self.r_pii_success = [
+                weblog.post("/login?auth=local", data={"username": self.USER, "password": self.PASSWORD}),
+                weblog.get("/login?auth=basic", headers={"Authorization": self.BASIC_AUTH_USER_HEADER}),
+            ]
+
+    def test_login_pii_success(self):
+        for r in self.r_pii_success:
+            assert r.status_code == 200
+            for _, _, span in interfaces.library.get_spans(request=r):
+                meta = span.get("meta", {})
+                assert meta["_dd.appsec.events.users.login.success.auto.mode"] == "safe"
+                assert meta["appsec.events.users.login.success.track"] == "true"
+                assert meta["usr.id"] == " "
+                assert meta["manual.keep"] == "true"
+
+    def setup_login_success(self):
+        self.library_name = context.library
+        self.r_success = []
+
+        if self.library_name == "nodejs":
+            self.r_success = [
+                weblog.post(
+                    "/login?auth=local",
+                    params={"auth": "local"},
```

Comment:
Aren't params duplicated? In query string and in `params` property

Reply:
Yes, removing it.